engine/lib/elgglib.php

```diff
@@ -1426,7 +1426,31 @@ function _elgg_normalize_plural_options_array($options, $singulars) {
  */
 function _elgg_shutdown_hook() {
 	global $START_MICROTIME;
+	
+	elgg_set_config('shutdown_flag', true);
+	
+	$headers = headers_list();
+	$connection_sent = false;
+	foreach ($headers as $hdr) {
+		if (stripos($hdr, "Connection") === 0) {
+			$connection_sent = true;
+		}
+	}
 
+	if (!headers_sent() && !$connection_sent) {
+		// Ignore user aborts and allow the script to run forever
+		ignore_user_abort(true);
+		// Prevent an APC session bug: https://bugs.php.net/bug.php?id=60657
+		session_write_close();
```

Comment:
We need docs that devs shouldn't write to session during shutdown. Better if there were a way to catch it.